x-pack/solutions/security/plugins/security_solution/public/data_view_manager/hooks/use_full_data_view.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useMemo, useState } from 'react';
+import {
+  type DataView,
+  type DataViewsServicePublic,
+  type DataViewSpec,
+} from '@kbn/data-views-plugin/public';
+import memoize from 'lodash/memoize';
+
+import { useKibana } from '../../common/lib/kibana';
+import { type DataViewManagerScopeName } from '../constants';
+import { useDataView } from './use_data_view';
+import { useIsExperimentalFeatureEnabled } from '../../common/hooks/use_experimental_features';
+
+/**
+ * Creates a DataView from the provided DataViewSpec.
+ *
+ * This is a memoized function that will return the same DataView instance across useFullDataView occurences in the project,
+ * for the same dataViewSpec version or title.
+ */
+const dataViewFromSpec: (
+  dataViewsService: DataViewsServicePublic,
+  dataViewSpec: DataViewSpec
+) => Promise<DataView | undefined> = memoize(
+  (dataViewsService: DataViewsServicePublic, dataViewSpec: DataViewSpec) => {
+    // NOTE: this should never happen for data views managed by the Data View Manager and is considered an error
+    if (!dataViewSpec.id) {
```

Comment:
What happens for adhoc dataviews?

Reply:
they have ids as well, prefixed with `adhoc_`